src/lib.rs

```diff
@@ -578,4 +578,97 @@ mod test {
         let result = select_coin_lowestlarger(&inputs, options);
         assert!(matches!(result, Err(SelectionError::InsufficientFunds)));
     }
+
+    #[test]
+    fn test_calculate_fee(){
+        let fee = calculate_fee(60, 5.0);
+        assert_eq!(fee, 300)
+    }
```

Comment:
While writing these type of arithmatic tests, its better if you go with some form of test vectors. Each case will test on various scenarios and more scenarios you can add more robust is your tests. Even for inputs with wrong data and assert that the functions can give the correct error message.

For example:

```
input: 60, 5, Output: 300.
input 60, 5.0, Output: 300.
input 60 , -5, Output: Invalid Fee rate
```
Etc. 

Same goes for other tests. You can start with a population of given inputs and expected outputs. And then the test just loops though each of them and assert that inputs matches with expected outputs.

Reply:
Thank you for your suggestions, I'm working on this now.  Ive got my tests working for inputs that are weight: u32 and rate: f32 which is what calculate_fee() takes as arguments.  In your examples it looks like I should test for cases where rate is given as a u32 and if so I'm unsure how to test that since the types don't match, the code wont compile.  Also In order to fix the calculate_fee() function so it returns an error when fee rate is negative I have to change the return type to a result and then call unwrap() everywhere in the code the function is called.  Am I heading in the right direction?

Reply:
Yes it's in the right direction. Don't take my example seriously, I didn't check the actual arg types while writing it. Its just for giving an idea.

Reply:
tests looks good. fmt is failing. Try to see if you can envision any more test edge cases. Or else this is good.

Reply:
I will think about more test edge cases.  Fmt is format?  Where can I learn more about how this test works?  Is there a way to see where my code fails?

Reply:
I added more edge cases and fixed the calculate_waste() and calculate_fee() functions to pass the new edge cases.  And I fixed the format.  Do you have any other suggestions?